code/__HELPERS/math/ray/ray.dm

```diff
@@ -32,6 +32,15 @@
 	direction = p_direction.chebyshev_normalized()
 	src.z = z
 
+/ray/Destroy()
+	qdel(origin)
```

Comment:
In this and the raycasts, are you sure you want to delete these? Is it specified that the vectors should never be reused after being used to create a ray or raycast?

Reply:
shit, you're right